source/jormungandr/jormungandr/instance.py

```diff
@@ -765,10 +766,30 @@ def get_autocomplete(self, requested_autocomplete):
     def get_ridesharing_journeys_with_feed_publishers(self, from_coord, to_coord, period_extremity, limit=None):
         res = []
         fps = set()
-        for service in self.ridesharing_services:
-            rsjs, fp = service.request_journeys_with_feed_publisher(
-                from_coord, to_coord, period_extremity, limit
-            )
-            res.extend(rsjs)
-            fps.add(fp)
+        greenlet_pool_actived = app.config.get('GREENLET_POOL_FOR_RIDESHARING_SERVICES', False)
+
+        with FutureManager() as future_manager:
+            futures = []
+            for service in self.ridesharing_services:
```

Comment:
```python
calls = []
res = []
fps = []

for service in self.ridesharing_services:
    def _call(s=service):
        return service.request_journeys_with_feed_publisher(from_coord, to_coord, period_extremity, limit)
    calls.append(_call)

call_res = []
if greenlet_pool_activated:
    with FutureManager() as future_manager:
        futures = [future_manager.create_future(call) for call in calls]
    # it's just a generator, a lazy call 
    call_res = (f.wait_and_get() for f in futures)
else:
    call_res = (c() for c in calls)

for rsjs, fp in call_res:
    res.extend(rsjs)
    fps.add(fp)      
        
return res, fps

Reply:
I think it didn't work.
I need to create future for all `service.request_journeys_with_feed_publisher(...)` call.
So when you do
```
for service in self.ridesharing_services:
    def _call(s=service):
        return service.request_journeys_with_feed_publisher(from_coord, to_coord, period_extremity, limit)
    calls.append(_call)
```
You call the function, and sequentially no ? Unless it return a functor ?

Reply:
`calls` contains a bunch of functors

Reply:
functors are triggered by:
```python
futures = [future_manager.create_future(call) for call in calls]
```

Disclaimer: I didn't test the code :p